client/state/selectors/get-current-user-localized-payment-methods.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { generateLangCcLocale } from 'state/i18n/util';
+import { getCurrentUserLocale } from 'state/current-user/selectors';
+import { getGeoCountryShort } from 'state/geo/selectors';
+
+/**
+ * Returns the preferred payment methods for the current locale.
+ *
+ * @param  {Object}  state       Global state tree
+ * @return {Array}               Preferred payment methods
+ */
+export function getCurrentUserLocalizedPaymentMethods( state ) {
```

Comment:
I went with the `getCurrent` paradigm on purpose so I'd lean toward your second suggestion. Yeah, I think that works fine.